components/html-block/html-block.js

```diff
@@ -201,32 +206,60 @@ class HtmlBlock extends LitElement {
 		return false;
 	}
 
-	_stamp(template) {
+	_findSlottedElement(tagName, slot) {
+		if (!slot) slot = this.shadowRoot.querySelector('slot');
+		return slot.assignedNodes({ flatten: true })
+			.find(node => (node.nodeType === Node.ELEMENT_NODE && node.tagName === tagName.toUpperCase()));
+	}
+
+	async _processRenderers(elem) {
+		for (const renderer of getRenderers()) {
+			if (this.noDeferredRendering && !renderer.canRenderInline) continue;
+
+			if (this._contextObserverController && renderer.contextAttributes) {
+				const contextValues = new Map();
+				renderer.contextAttributes.forEach(attr => contextValues.set(attr, this._contextObserverController.values.get(attr)));
+				elem = await renderer.render(elem, contextValues);
+			} else {
+				elem = await renderer.render(elem);
+			}
+		}
+
+		return elem;
+	}
+
+	async _render(slot) {
+		if (this.noDeferredRendering) await this._renderInline(slot);
+		else this._stamp(slot);
+	}
+
+	async _renderInline(slot) {
+		const noDeferredRenderingContainer = this._findSlottedElement('DIV', slot);
+		if (!noDeferredRenderingContainer) return;
+		await this._processRenderers(noDeferredRenderingContainer);
+	}
+
+	_stamp(slot) {
 		const stampHTML = async template => {
 			const fragment = template ? document.importNode(template.content, true) : null;
 			if (fragment) {
 
 				let temp = document.createElement('div');
 				temp.appendChild(fragment);
 
-				for (const renderer of getRenderers()) {
-					if (this._contextObserverController && renderer.contextAttributes) {
-						const contextValues = new Map();
-						renderer.contextAttributes.forEach(attr => contextValues.set(attr, this._contextObserverController.values.get(attr)));
-						temp = await renderer.render(temp, contextValues);
-					} else {
-						temp = await renderer.render(temp);
-					}
-				}
+				temp = await this._processRenderers(temp);
 				this._renderContainer.innerHTML = temp.innerHTML;
 
 			} else {
 				this._renderContainer.innerHTML = '';
 			}
 		};
 
+		const template = this._findSlottedElement('TEMPLATE', slot);
+
 		if (this._templateObserver) this._templateObserver.disconnect();
 		if (template) {
+			this._noDeferredRendering = false;
```

Comment:
I think this isn't needed anymore, correct?

Reply:
Oops, you're right. Missed this spot.